### PR TITLE
SFTP client: listFiles works with >97 files

### LINKED ISSUE #16: SFTP client does not download all available files from remote server directory.
I have in my directory 144 files. This test will download only 98 files:

```
  it('should list more than 97 files in remote folder', function (done) {
    this.timeout(60000)
    sftp.listFiles("my/source/folder")
    .then (files => {
      expect(files.length).toBe(144)
      done()
    })
    .catch(done)
  })


  1) SFTP Helper Integration Tests should list more than 97 files in remote folder:
     Error: Expected 98 to be 144
```

This is probably related to this issue: https://github.com/mscdex/ssh2/issues/274
Also with newer version one does not need to `opendir` anymore (https://github.com/sphereio/sphere-node-utils/blob/master/src/coffee/helpers/sftp.coffee#L21) and can work directly with ´´´readdir´´´ which returns all files.

### PR DESCRIPTION
**Pull request by dhettler**
Fixes #16

### CODE REVIEW

File: src/coffee/helpers/sftp.coffee [hunk 1]
```diff
@@ -18,19 +18,11 @@ class Sftp
   ###
   listFiles: (sftp, dirName) ->
     new Promise (resolve, reject) ->
-      sftp.opendir dirName, (err, handle) ->
-        if err
+      sftp.readdir dirName, (err, list) ->
+        if err?
```
Comment (butenkor) [inline src/coffee/helpers/sftp.coffee]: why ```err``` is not enough and you do ```err?```?

Comment (dhettler) [inline src/coffee/helpers/sftp.coffee]: doesn't make a huge difference as I can't think of a reason why err should be `0`, changed it anyway: f04467a

**Comment by butenkor** [comment]
@dhettler Please provide test

**Comment by butenkor** [comment]
https://github.com/sphereio/sphere-node-utils/pull/23

**Comment by dhettler** [comment]
The only sensible way to test this change is through an integration test. Is there an SFTP server for integration tests?

**Comment by hisabimbola** [comment]
Yes, there is @butenkor should be able to provide you with one else you can ask ops

**Comment by butenkor** [comment]
There are already integration tests which use test SFTP account taken from travis OS env. See:

https://github.com/sphereio/sphere-node-utils/blob/master/.travis.yml#L6
and
https://github.com/sphereio/sphere-node-utils/blob/master/src/spec/helpers/sftp.spec.coffee#L6

**Comment by dhettler** [comment]
@butenkor @hisabimbola added test
https://github.com/sphereio/sphere-node-utils/pull/23/files#diff-71d230127412bbf5af9d5d2747f2529fR103

**Comment by butenkor** [comment]
@hisabimbola @PhilippSpo please approve and release :)

**Pull request merged**